providers/aws-ec2/src/test/java/org/jclouds/aws/ec2/features/AWSSubnetApiMockTest.java

```diff
@@ -0,0 +1,88 @@
+      enqueue(DEFAULT_REGION, new MockResponse().setBody("<CreateSubnetResponse xmlns=\"http://ec2.amazonaws.com/doc/2016-09-15/\">\n" +
+              "  <requestId>7a62c49f-347e-4fc4-9331-6e8eEXAMPLE</requestId>\n" +
+              "  <subnet>\n" +
+              "    <subnetId>subnet-9d4a7b6c</subnetId>\n" +
+              "    <state>pending</state>\n" +
+              "    <vpcId>vpc-1a2b3c4d</vpcId>\n" +
+              "    <cidrBlock>10.0.1.0/24</cidrBlock> \n" +
+              "    <availableIpAddressCount>251</availableIpAddressCount>\n" +
+              "    <availabilityZone>us-east-1a</availabilityZone>\n" +
+              "    <tagSet/>\n" +
+              "  </subnet>\n" +
+              "</CreateSubnetResponse>"));
+
+      Subnet result = subnetApi().createSubnetInRegion(DEFAULT_REGION, "vpc-1a2b3c4d", "10.0.1.0/24");
+      assertEquals(result.getVpcId(), "vpc-1a2b3c4d");
+      assertEquals(result.getCidrBlock(), "10.0.1.0/24");
+      assertEquals(result.getAvailabilityZone(), "us-east-1a");
+      assertEquals(result.getSubnetId(), "subnet-9d4a7b6c");
+      assertEquals(result.getSubnetState().value(), "pending");
+
+      assertPosted(DEFAULT_REGION, "Action=DescribeRegions");
+      assertPosted(DEFAULT_REGION, "Action=CreateSubnet&VpcId=vpc-1a2b3c4d&CidrBlock=10.0.1.0/24");
+   }
+
+   public void createSubnetInRegion_options() throws Exception {
+      enqueueRegions(DEFAULT_REGION);
+      enqueue(DEFAULT_REGION,
+            new MockResponse().setBody("<CreateSubnetResponse xmlns=\"http://ec2.amazonaws.com/doc/2016-09-15/\">\n" +
+                    "  <requestId>7a62c49f-347e-4fc4-9331-6e8eEXAMPLE</requestId>\n" +
+                    "  <subnet>\n" +
+                    "    <subnetId>subnet-9d4a7b6c</subnetId>\n" +
+                    "    <state>pending</state>\n" +
+                    "    <vpcId>vpc-1a2b3c4d</vpcId>\n" +
+                    "    <cidrBlock>10.0.1.0/24</cidrBlock> \n" +
+                    "    <availableIpAddressCount>251</availableIpAddressCount>\n" +
+                    "    <availabilityZone>us-east-1a</availabilityZone>\n" +
+                    "    <tagSet/>\n" +
+                    "  </subnet>\n" +
+                    "</CreateSubnetResponse>"));
+
+      Subnet result = subnetApi().createSubnetInRegion(DEFAULT_REGION, "vpc-1a2b3c4d", "10.0.1.0/24",
+            new CreateSubnetOptions().dryRun().availabilityZone("us-east-1a"));
+      assertEquals(result.getVpcId(), "vpc-1a2b3c4d");
+      assertEquals(result.getCidrBlock(), "10.0.1.0/24");
+      assertEquals(result.getAvailabilityZone(), "us-east-1a");
+      assertEquals(result.getSubnetId(), "subnet-9d4a7b6c");
+      assertEquals(result.getSubnetState().value(), "pending");
+      
+      assertPosted(DEFAULT_REGION, "Action=DescribeRegions");
+      assertPosted(DEFAULT_REGION, "Action=CreateSubnet&VpcId=vpc-1a2b3c4d&CidrBlock=10.0.1.0/24&DryRun=true&AvailabilityZone=us-east-1a");
+   }
```

Comment:
Add the tests for the `describe` methods as well, and add the tests that exercise the defined fallbacks too.